components/dashboard/src/utils.ts

```diff
@@ -230,3 +233,53 @@ export function isTrustedUrlOrPath(urlOrPath: string) {
     }
     return isTrusted;
 }
+
+type UnifiedAuthProvider = "Bitbucket" | "GitLab" | "GitHub" | "Azure DevOps";
+
+const isIdentity = (identity?: AuthProviderDescription): identity is AuthProviderDescription => !!identity;
```

Comment:
Hm, not sure I understand this function? It does not check the shape of AuthProviderDescription, so not sure this should be an `is` function? And it does not check identity, really, either... :thinking:

Reply:
Ah, just seeing it's a move.
Might still be worth the cleanup if we have the context right now.

Reply:
Nice catch! Made sense to clean up, cause since upgrading to [TS 5.5](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html), we're able to do this without these `is` functions and just do this:
```js
const userProviders = userIdentities
    .map((id) => descriptions?.find((provider) => provider.id === id))
    .filter(p => !!p)
```

Reply:
Pushed [`0147dd7` (#20415)](https://github.com/gitpod-io/gitpod/pull/20415/commits/0147dd7c4d083f13713abdfa8513ae08d70bff66) with the changes.